bpf/lib/host_firewall.h

```diff
@@ -138,11 +136,11 @@ ipv6_host_policy_ingress(struct __ctx_buff *ctx, __u32 *src_id)
 		return ret;
 
 	/* Retrieve source identity. */
-	info = lookup_ip6_remote_endpoint(&orig_sip);
+	info = lookup_ip6_remote_endpoint(&tuple.saddr);
```

Comment:
I don't think that is correct. Depending on the return code (e.g., `CT_NEW`), `ct_lookup6` may call `ipv6_ct_tuple_reverse`, which may result in a reversed tuple where `tuple.saddr` actually is the destination address.

It might however be possible to use `ip6->saddr` directly, with a cast.

Reply:
Ah, I missed that path in `ct_lookup6`. My impression was that when doing `ct_lookup6` we don't modify tuple's source and destination address. Thanks for noticing this. :rocket: 
It makes sense to use `ip6->saddr` directly here.